benchmarks/bench_leaf_test.go

```diff
@@ -0,0 +1,82 @@
+package smt
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"github.com/pokt-network/smt"
+	"github.com/stretchr/testify/require"
+	"strconv"
+	"testing"
+)
+
+func BenchmarkSMTLeafSizes_Fill(b *testing.B) {
+	treeSizes := []int{100000, 500000, 1000000, 5000000, 10000000}
+	leafSizes := []int{256, 512, 1024, 2048, 4096, 8192, 16384}
+	nodes, err := smt.NewKVStore("")
+	require.NoError(b, err)
+	for _, treeSize := range treeSizes {
+		for _, leafSize := range leafSizes {
+			leaf := make([]byte, leafSize)
+			for i := 0; i < 2; i++ {
+				commit := i == 1
+				name := "Fill"
+				if commit {
+					name = "Fill & Commit"
+				}
+				tree := smt.NewSparseMerkleTree(nodes, sha256.New(), smt.WithValueHasher(nil))
+				b.ResetTimer()
+				b.Run(
+					fmt.Sprintf("%s [Leaf Size: %d bytes] (%d)", name, leafSize, treeSize),
+					func(b *testing.B) {
+						b.ResetTimer()
+						b.ReportAllocs()
+						for i := 0; i < treeSize; i++ {
+							require.NoError(b, tree.Update([]byte(strconv.Itoa(i)), leaf))
+						}
+						if commit {
```

Comment:
Do we have a test where you commit after every update and compare that?

Don't do it in this commit but leave a comment saying: `// There is an opportunity to do a fuzz test where we commit every N updates, if this ever becomes a bottlneck`